testutils/zk_utils.go

```diff
@@ -54,7 +54,9 @@ func StartZookeeper(opts ...func(*ZKConfig)) (*ZkControl, error) {
 
 	// the container IP is not routable on Darwin, thus needs port
 	// mapping for the container.
-	hostConfig := &container.HostConfig{}
+	hostConfig := &container.HostConfig{
+		AutoRemove: true,
```

Comment:
instead of hard-coding this, maybe make it a field in `ZKConfig` and then you can configure it using a functional option. I'd like to avoid hard-coding new things here.